### PR TITLE
Upgraded to Glide 4.9.0

### PR DESCRIPTION
**Pull request by rubo**
This PR addresses the issue #464.
As v4.8.0 brought some serious changes, there are a lot of them in bindings too. I also took some cool ideas from @jonathanpeppers' glidex.

I'd like to stress the following changes:

1. Two Glide dependencies (GIF Decoder and Disk LRU Cache) have been added as separate packages since they now come in separate .aar and .jar respectively.
I added them as dependencies in NuGet spec of Xamarin.Android.Glide. However, to make it work, we need to manually add them directly to the main project where Glide is used. Otherwise, there's a runtime error as [docs say](https://bumptech.github.io/glide/doc/download-setup.html#download). I'm not sure whether I need to reference them from Xamarin.Android.Glide project too. Doing so doesn't fix the aforementioned error nor it's needed at compile time.

2. Unified project files and NuGet specs. Updated project files.

3. The original Glide package has a dependencies on Android Support Fragment and Animated Vector Drawable. However, in our NuGet spec we have dependencies on Android Support v4 and Annotations. I'm not sure what to do here.

Once done with this PR, I'd like to add more integration libraries for Glide like [OkHttp3](https://bumptech.github.io/glide/int/okhttp3.html) and [RecyclerView](https://bumptech.github.io/glide/int/recyclerview.html).

### CODE REVIEW

File: Android/Glide/source/Xamarin.Android.Glide/Additions/Additions.cs [hunk 1]
```diff
@@ -1,417 +1,412 @@
...
 		}
-
 	}
 }
 
+namespace Bumptech.Glide.Load.Engine.Executor
+{
+    public partial class GlideExecutor
+    {
+        IList IExecutorService.InvokeAll(ICollection tasks) => InvokeAll(tasks.Cast<ICallable>().ToList()).ToList();
```
Comment (rubo) [inline Android/Glide/source/Xamarin.Android.Glide/Additions/Additions.cs:185]: I'm not sure this is the best way for `ICollection` to `ICollection<T>` and `IList<T>` to `IList` conversion.

**Comment by xyfoo** [comment]
@rubo  I've implemented the RecyclerView integration library v4.0 a few days ago (https://github.com/xyfoo/Bumptech.Glide.Integration.RecyclerView). Not sure if that would be useful to you.

**Comment by rubo** [comment]
@xyfoo Thanks for sharing. I see you had a problem with bindings and went with pure C# instead. Nevertheless, it should be a binding project. I'll check once I'm done with this.

**Comment by Redth** [comment]
Thanks, I merged this into another branch and just going to clean up a few minor things, the contribution is much appreciated!

**Pull request merged**